lib/datastore/dataset.js

```diff
@@ -132,24 +132,27 @@ nodeutil.inherits(Dataset, DatastoreRequest);
  *
  * @private
  *
- * @param {object} options - Configuration object.
- * @param {string=} options.apiEndpoint - Custom API endpoint.
+ * @param {string} customApiEndpoint - Custom API endpoint.
  */
-Dataset.determineApiEndpoint_ = function(options) {
-  var apiEndpoint = 'https://www.googleapis.com';
+Dataset.prototype.determineApiEndpoint_ = function(customApiEndpoint) {
```

Comment:
Testing this method by itself wasn't bad, but testing how it integrates with the constructor wasn't great. I much prefer the no-side effect route, but couldn't find a way to easily fit that in here. We not only need "what is the API endpoint?" answered, but "is it the default one or a custom one?" Ideally, one function can answer both of those questions, so I could either return an object like `{ isCustom: false, apiEndpoint: 'http://...' }` or just treat this method as an extension of the initialization process, giving it full control of setting instance variables. (Note the method moved from static to instance)

I'm not a big fan of either approach, really. But, we have two **major** overhauls to Datastore coming soon (v1beta3 upgrade which is a complete re-write (#897), and after that lands, we will implement the `getOrCreate` base classes, "Service", and "ServiceObject"). Those will probably/definitely force a new solution to this problem fit in with the changes.